pkg/ebpf/c/tracee.bpf.c

```diff
@@ -372,13 +372,14 @@ Copyright (C) Aqua Security inc.
     BPF_MAP(_name, BPF_MAP_TYPE_PERF_EVENT_ARRAY, int, __u32, 1024)
 
 // stack traces: the value is 1 big byte array of the stack addresses
+typedef __u64 stack_trace_t[MAX_STACK_DEPTH];
 #define BPF_STACK_TRACE(_name, _max_entries)                            \
-    struct bpf_map_def SEC("maps") _name = {                            \
-        .type = BPF_MAP_TYPE_STACK_TRACE,                               \
-        .key_size = sizeof(u32),                                        \
-        .value_size = sizeof(size_t) * MAX_STACK_DEPTH,                 \
-        .max_entries = _max_entries,                                    \
-    };
+    struct {                                                            \
+        __uint(type, BPF_MAP_TYPE_STACK_TRACE);                         \
+        __uint(max_entries, _max_entries);                              \
+        __type(key, u32);                                               \
+        __type(value, stack_trace_t);                                   \
+    } _name SEC(".maps");
```

Comment:
As you defined stack_trace_t above, you can now simply use the BPF_MAP macro:
`BPF_MAP(_name, BPF_MAP_TYPE_STACK_TRACE, u32, stack_trace_t, _max_entries)`

Reply:
Good call